cfme/tests/cloud_infra_common/test_relationships.py

```diff
@@ -134,3 +138,60 @@ def test_cloud_provider_relationships(appliance, provider, setup_provider, relat
     provider_view.entities.relationships.click_at(relationship)
     relationship_view = appliance.browser.create_view(view, additional_context={'object': obj})
     assert relationship_view.is_displayed
+
+
+@pytest.fixture(scope='function')
+def prov_child_visibility(appliance, provider, request, tag, user_restricted):
+    def _prov_child_visibility(relationship, view, visibility):
+        provider.add_tag(tag=tag)
+
+        @request.addfinalizer
+        def _finalize():
+            provider.remove_tag(tag=tag)
+
+        with user_restricted:
+            try:
+                provider_view = navigate_to(provider, "Details")
+                if provider.get_detail("Relationships", relationship) == "0":
+                    pytest.skip("There are no relationships for {}".format(relationship))
+                provider_view.entities.relationships.click_at(relationship)
+                relationship_view = appliance.browser.create_view(view)
+                assert relationship_view.entities.entity_names
+                actual_visibility = True
+            except Exception:
+                actual_visibility = False
+
+        assert actual_visibility == visibility
+    return _prov_child_visibility
+
+
+@pytest.mark.parametrize("relationship,view", INFRA_PROVIDER_RELATIONSHIPS,
+    ids=[rel[0] for rel in INFRA_PROVIDER_RELATIONSHIPS])
+@pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE)
```

Comment:
**Required**: Use ONE_PER_CATEGORY here, in such case only cloud or infra will be added to test name, and not the provider name